fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -104,15 +104,19 @@
     private final Cache<String, FDBLuceneFileReference> fileReferenceCache;
     private final Cache<Pair<Long, Integer>, CompletableFuture<byte[]>> blockCache;
 
-    public FDBDirectory(@Nonnull Subspace subspace, @Nonnull FDBRecordContext context) {
-        this(subspace, context, NoLockFactory.INSTANCE);
+    private final boolean compressionEnabled;
+    private final boolean encryptionEnabled;
+
+    public FDBDirectory(@Nonnull Subspace subspace, @Nonnull FDBRecordContext context, boolean compressionEnabled, boolean encryptionEnabled) {
+        this(subspace, context, compressionEnabled, encryptionEnabled, NoLockFactory.INSTANCE);
     }
 
-    FDBDirectory(@Nonnull Subspace subspace, @Nonnull FDBRecordContext context, @Nonnull LockFactory lockFactory) {
-        this(subspace, context, lockFactory, DEFAULT_BLOCK_SIZE, DEFAULT_INITIAL_CAPACITY, DEFAULT_MAXIMUM_SIZE, DEFAULT_CONCURRENCY_LEVEL);
+    FDBDirectory(@Nonnull Subspace subspace, @Nonnull FDBRecordContext context, boolean compressionEnabled, boolean encryptionEnabled, @Nonnull LockFactory lockFactory) {
+        this(subspace, context, compressionEnabled, encryptionEnabled, lockFactory, DEFAULT_BLOCK_SIZE, DEFAULT_INITIAL_CAPACITY, DEFAULT_MAXIMUM_SIZE, DEFAULT_CONCURRENCY_LEVEL);
     }
 
-    FDBDirectory(@Nonnull Subspace subspace, @Nonnull FDBRecordContext context, @Nonnull LockFactory lockFactory, int blockSize, final int initialCapacity, final int maximumSize, final int concurrencyLevel) {
+    FDBDirectory(@Nonnull Subspace subspace, @Nonnull FDBRecordContext context, boolean compressionEnabled, boolean encryptionEnabled, @Nonnull LockFactory lockFactory,
```

Comment:
It feels weird that `compressionEnabled` and `encryptionEnabled` are passed in as constructor arguments rather than reading them from the record context here. (Memoizing their values as members of the class rather than re-reading them from the context properties each time seems reasonable, though.)

Reply:
Yeah I would prefer to read the prop value in constructor so it won't be loaded from prop every time. Especially considering loading prop from the supplier may not be free (it could be from property file). Btw, this reminds me that I should have a cacheing value for ```FDBPropertyKey``` to avoid duplicate loading of value from supplier. I will make this change in next commit. But you have a good point that no need to pass in these 2 parameters from ```DirectoryCommitCheckAsync``` level. So I will make a change to load the prop values here rather than higher level.

Reply:
Done.